include/ck_tile/core/config.hpp

```diff
@@ -280,3 +280,151 @@
 #ifndef CK_TILE_ENC_SUPPORT_Y_TO_R
 #define CK_TILE_ENC_SUPPORT_Y_TO_R 0
 #endif
+
+// Defines CK_TILE specific architecture symbols
+// Guarantee the following symbols are defined to either 0 or 1:
+// CK_TILE_ARCH_GFX908
+// CK_TILE_ARCH_GFX90A
+// CK_TILE_ARCH_GFX942
+// CK_TILE_ARCH_GFX950
+// CK_TILE_ARCH_GFX1100
+// CK_TILE_ARCH_GFX1101
+// CK_TILE_ARCH_GFX1102
+// CK_TILE_ARCH_GFX1151
+// CK_TILE_ARCH_GFX1200
+// CK_TILE_ARCH_GFX1201
+// CK_TILE_ARCH_HOST
+// TODO: Replace specific instances of __gfxXYZW___ with CK_TILE_ARCH_GFXXYZW
+#if defined(__HIP_DEVICE_COMPILE__) && __HIP_DEVICE_COMPILE__
```

Comment:
This may just be historic, but I'm finding this mix of preprocessor constants and C++ constexpr values confusing. Wouldn't it be cleaner to just capture the raw state preprocssor in one constexpr struct, then do all this logic in c++ constexpr functions and values. Right now we have a lot of preprocessor definitions, then we translate to C++-friendly functions. It doesn't need to be changed now, but it might simplify the code to capture state clearly: the state during the compile pass (convert to c++ constexpr), the state of CMake flags (as constexpr), the fundamental state, and any erived state. I like the functionality of what you have, it just seems like it could be simplified further by isolating the #ifdefs into c++ constexpr contexts for compiler-pass options and CMake options.